### PR TITLE
Add new consumer for searching

### PR DESCRIPTION
**Pull request by BelehovEgor**

### CODE REVIEW

File: src/UserService.Data/UserRepository.cs [hunk 1]
```diff
@@ -203,5 +203,12 @@ public bool IsExistUser(Guid userId)
         {
             return _provider.Users.FirstOrDefault(u => u.Id == userId) != null;
         }
+
+        public IEnumerable<DbUser> Search(string text)
+        {
+            return _provider.Users.Where(u => u.FirstName.Contains(text, StringComparison.OrdinalIgnoreCase)
+                                            || u.MiddleName.Contains(text, StringComparison.OrdinalIgnoreCase)
```
Comment (djino97) [inline src/UserService.Data/UserRepository.cs:210]: Last name search more often then other name. Then must search firstName, next -- MiddleName

File: test/UserService.Broker.UnitTests/Consumers/SearchUsersConsumerTests.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            };
+
+            _repository = new();
+            _repository
+                .Setup(r => r.Search(ExistName))
+                .Returns(_dbUsers);
+        }
+
+        [Test]
+        public async Task ShouldConsumeSuccessful()
```
Comment (djino97) [inline test/UserService.Broker.UnitTests/Consumers/SearchUsersConsumerTests.cs:75]: strange test name

**Review by djino97 — APPROVED** [review]

File: src/UserService.Broker/Consumers/SearchUsersConsumer.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        private IUserRepository _userRepository;
+
+        private object SearchUsers(string text)
+        {
+            List<DbUser> users = _userRepository.Search(text).ToList();
+
+            return ISearchResponse.CreateObj(
+                users.Select(
+                    u => new SearchInfo(u.Id, string.Join(" ", u.LastName, u.FirstName))).ToList());
```
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/UserService.Broker/Consumers/SearchUsersConsumer.cs:25]: maybe, but I think it is not better than this

File: test/UserService.Broker.UnitTests/Consumers/SearchUsersConsumerTests.cs (hunk 2 above)
Comment (BelehovEgor) [reply test/UserService.Broker.UnitTests/Consumers/SearchUsersConsumerTests.cs:75]: fixed

File: src/UserService.Data/UserRepository.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/UserService.Data/UserRepository.cs:210]: fixed

**Review by MariyaSin — APPROVED** [review]

**Review by TIove — APPROVED** [review]

**Pull request merged**